documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -133,3 +133,22 @@ The date MAY optionally be followed by one of:
 * 2016-07-04-rc0 - Just use "rc", not "rc" + number
 
 Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="M3016" />M3016 PropertiesNamesCamelCase
+**Output Message**: Property named: "{0}", must follow camelCase style. Example: "{1}".
+**Output Message**: Property named: "{0}", for definition: "{1}" must follow camelCase style. Example: "{2}".
+
+**Description**: Property names must use lowerCameCase style.  
+
+**Good Examples**: Examples of lowerCamelCase style:
+* camelCase
+* resourceKey
+* fooBarBaz
+
+**Bad Examples**: The following would be invalid:
+* PascalCase
+* UpperCamelCase
+* alllowercase
+* miXeDcApItAlIzAtIoN
```

Comment:
the rule would consider this valid, even though looking at it, seems to make no sense to do that, the rule is not violated as it does not recognize semantic words, but mostly checks there are no more than 3 upper case letters together - the reason for this is, we need to allow acronyms like IP and if another word comes after it can be capitalized too, so publicIPAdress is a valid property name per this rule.